spec/models/manageiq/providers/hawkular/middleware_manager_spec.rb

```diff
@@ -227,4 +227,107 @@ def timeline_domain_expectations(status)
+          :credentials => { :username => username, :password => password },
+          :options     => { :tenant => 'hawkular', :verify_ssl => verify_ssl, :ssl_cert_store => ssl_cert_store }
+        }
+      end
+      subject { ems.connect }
+
+      before do
+        allow(ems).to receive_messages(
+          :hostname                => host,
+          :port                    => port,
+          :default_endpoint        => endpoint,
+          :authentication_userid   => username,
+          :authentication_password => password
+        )
+      end
+
+      context 'non-ssl' do
+        let(:http_scheme) { URI::HTTP }
+
+        it { is_expected.to be_connected_to connection_options }
+      end
+
+      context 'ssl-without-validation' do
+        let(:verify_ssl) { 0 }
+        let(:endpoint)   { FactoryGirl.create(:endpoint, :security_protocol => 'ssl-without-validation') }
+
+        it { is_expected.to be_connected_to connection_options }
+      end
+
+      context 'ssl' do
+        let(:endpoint) { FactoryGirl.create(:endpoint, :security_protocol => 'ssl') }
+
+        it { is_expected.to be_connected_to connection_options }
+      end
+    end
+
+    context 'validation' do
+      shared_examples 'validate' do
+        let(:connection) { dup }
+        let(:struct) { dup }
+        before do
+          allow(connection).to receive(:inventory).and_return(struct)
+          allow(ems).to receive(:connect).and_return(connection)
+        end
+
+        context 'failed' do
+          before do
+            allow(struct).to receive(:list_feeds).and_raise(exception)
+          end
+
+          [
+            [::Hawkular::Exception.new("Unauthorized", 401), MiqException::MiqInvalidCredentialsError, /Invalid credentials/],
+            [::Hawkular::Exception.new("Host not found", 404), MiqException::MiqHostError, /Hawkular not found on host/],
+            [::Hawkular::ConnectionException.new(nil), MiqException::MiqUnreachableError, /Unable to connect to*/],
+            [URI::InvalidComponentError, MiqException::MiqHostError, /Host .* is invalid/],
+            [StandardError, MiqException::Error, /Unable to verify credentials/]
+          ].each do |raised, expectation, message|
+            context message do
+              let(:exception) { raised }
+
+              it 'raises exception' do
+                expect { subject }.to raise_error(expectation, message)
+              end
+            end
+          end
+        end
+
+        context 'sucessfull' do
+          before do
+            allow(struct).to receive(:list_feeds).and_return([])
+          end
+
+          it { is_expected.to be_truthy }
+        end
+      end
+
+      describe '#verify_credentials' do
+        subject { ems.verify_credentials }
+        include_examples 'validate'
+      end
+
+      describe '.raw_connect' do
+        subject { described_class.raw_connect(host, port, username, password, nil, nil, true) }
```

Comment:
I looked at my previous comments and I asked for a test where you pass `false` in the last param to check that no validations are done... Not sure if it's a useful test, though.

Reply:
I don't think we need such test since it's not doing any validation with `false`. That would only check if we can create a client instance from given params. Which doesn't make sense to test here.

Reply:
I know, but it's a behavior to cover (code is already covered). How are you ensuring that by passing false, or passing no parameter at all, the validation won't happen?
Although, knowing how this is used, it's really not very relevant. So, it's up to you.